patches/minecraft/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -53,7 +53,16 @@
              }, ChatType.CHAT, this.f_9743_.m_142081_());
           }
  
-@@ -1222,6 +_,7 @@
+@@ -1216,12 +_,14 @@
+             return;
+          }
+ 
+-         double d0 = 36.0D;
+-         if (this.f_9743_.m_20280_(entity) < 36.0D) {
++         double maxRange = this.f_9743_.m_21133_(net.minecraftforge.common.ForgeMod.ATTACK_RANGE.get()) + 3; //Forge: Change constant value to max possible range. Equal to current value + 3 (because creative gives +3)
++         maxRange = Math.max(maxRange, this.f_9743_.m_21133_(net.minecraftforge.common.ForgeMod.REACH_DISTANCE.get()) + 0.5); //Used for both attacks and interacts, needs to be the highest of the two for sanity.
```

Comment:
Shouldn't you use ServerboundInteractPacket.Action to determine if it is an attack action and only then use the attack range as the max range? Because atm this tightly couples both attack range and block reach together, meaning to properly validate the attack range on the server side you have to change both. (else the client could potentially send attack packets from further away)

Also shouldn't you test first if the player is in creative before applying the +3?

Reply:
The edit to this check was to mirror vanilla (it checks that the distance is within the absolute maximum possible).
I suppose further sanity might be desirable.

Reply:
Depends on how paranoid of clients we want to be and how network delay (in multiplayer) impacts this. I could imagine it will be harder to hit fast moving players/mobs with stricter range checks.

Reply:
Checks are now done at a lower level - thus validation should be a bit better.